Lib/http/client.py

```diff
@@ -1085,18 +1085,17 @@ def putrequest(self, method, url, skip_host=False,
         else:
             raise CannotSendRequest(self.__state)
 
-        # Save the method we use, we need it later in the response phase
+        # Save the method for use later in the response phase
         self._method = method
-        if not url:
-            url = '/'
-        # Prevent CVE-2019-9740.
-        if match := _contains_disallowed_url_pchar_re.search(url):
-            raise InvalidURL(f"URL can't contain control characters. {url!r} "
-                             f"(found at least {match.group()!r})")
-        request = '%s %s %s' % (method, url, self._http_vsn_str)
+
+        request = b' '.join((
+            method.encode('ascii'),
+            self._prepare_path(url or '/'),
+            self._http_vsn_str.encode('ascii')
+        ))
 
         # Non-ASCII characters should have been eliminated earlier
```

Comment:
We can get rid of this comment now.  it was referring to the `.encode('ascii')` that is no longer below.